src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -34,8 +34,9 @@
 import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
 
 /**
- * Checks that particular class are never used as types in variable
- * declarations, return values or parameters.
+ * Checks that particular classes are never used as types in variable declarations,
+ * one-dimensional and multi-dimensional arrays, return values, parameters
+ * or type parameters/arguments in method calls/references.
```

Comment:
shouldn't this also include type parameters of fields?

Reply:
No, it does not validate fields. Maybe this should be changed.